src/test/java/software/amazon/awssdk/crt/test/Mqtt5ClientTest.java

```diff
@@ -2512,5 +2512,133 @@ public void Retain_UC1() {
         }
     }
 
+    /**
+     * ============================================================
+     * Operation Interrupt Tests
+     * ============================================================
+     */
+
+    // Subscribe interrupt test
+    @Test
+    public void Interrupt_Sub_UC1() {
+        skipIfNetworkUnavailable();
+        Assume.assumeTrue(mqtt5DirectMqttHost != null);
+        Assume.assumeTrue(mqtt5DirectMqttPort != null);
+        int messageCount = 10;
+        String testUUID = UUID.randomUUID().toString();
+        String testTopic = "test/MQTT5_Binding_Java_" + testUUID;
+
+        try {
+            Mqtt5ClientOptionsBuilder builder = new Mqtt5ClientOptionsBuilder(mqtt5DirectMqttHost, mqtt5DirectMqttPort);
+            LifecycleEvents_Futured events = new LifecycleEvents_Futured();
+            builder.withLifecycleEvents(events);
+            Mqtt5Client client = new Mqtt5Client(builder.build());
```

Comment:
Havent looked at the other tests, but from an example standpoint, I would much prefer to see the client built in a try-with-resources block, dropping the explicit close at the end.  The way it's currently done, any exception in the try block leaks the client.

Reply:
Done! Now all CrtResources use try-with-resources blocks. The only exceptions are the overflow tests, which would have been really messy to convert (and they shouldn't have usable CrtResources that could leak anyway).